prosody-plugins/token/util.lib.lua

```diff
@@ -38,19 +29,12 @@ local function _verify_token(token, appId, appSecret, roomName, disableRoomNameC
 	if roomClaim == nil and disableRoomNameConstraints ~= true then
 		return nil, "'room' claim is missing";
 	end
-	if roomName ~= nil and roomName ~= roomClaim and disableRoomNameConstraints ~= true then
-		return nil, "Invalid room name('room' claim)";
-	end
 
 	return true;
 end
 
-function _M.verify_token(token, appId, appSecret, roomName, disableRoomNameConstraints)
-	return _verify_token(token, appId, appSecret, roomName, disableRoomNameConstraints);
-end
-
-function _M.get_room_name(token, appSecret)
```

Comment:
Why get_room_name was removed ? It looks like part of the logic related to room verification is now broken, see: https://github.com/SamWhited/jitsi-meet/blob/7f2fa9597cd43e2b54ead91a65cfe4a9e22c2505/prosody-plugins/mod_auth_token.lua#L146

Since now we need "kid" for ASAP it would make sense to add a method that decodes the token. Then use it to read both 'room' and 'kid' claims. Especially that the library supports it - just skip the secret or 3rd argument in jwt.decode:
https://github.com/paweldomas/luajwt/blob/master/luajwtjitsi.lua#L106